lib/Wikia/src/Service/User/Attributes/UserAttributes.php

```diff
@@ -141,7 +141,10 @@ private function attributeShouldBeSaved( $name, $value ) {
 	}
 
 	private function attributeShouldBeDeleted( $name, $value ) {
-		return isset( $this->defaultAttributes[$name] ) && $value == $this->defaultAttributes[$name];
+		return (
+			( isset( $this->defaultAttributes[$name] ) && $value == $this->defaultAttributes[$name] ) ||
+			is_null( $value )
```

Comment:
Not too familiar with attributes logic, but could this potentially conflict with the logic here https://github.com/Wikia/app/blob/SOC-1820/lib/Wikia/src/Service/User/Attributes/UserAttributes.php#L103-L105 if the default attribute for the attribute in question happens to be non-null? The `$value` parameter seems to be populated with the value of an attribute object loaded by `loadAttributes()`.